backend/base/models.py

```diff
@@ -1,3 +1,54 @@
 from django.db import models
-
+from django.contrib.auth.models import User
 # Create your models here.
+
+
+class Product(models.Model):
+    name = models.CharField(max_length=255)
+    description = models.TextField()
+    image = models.ImageField()
+    price = models.DecimalField(max_digits=7, decimal_places=2)
+    count_in_stock = models.IntegerField(default=0)
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    def __str__(self):
+        return self.name
+
+
+class Order(models.Model):
+    user = models.ForeignKey(User,on_delete=models.SET_NULL, null=True)
+    payment_method = models.CharField(max_length=255)
+    shipping_price = models.DecimalField(max_digits=7, decimal_places=2)
+    total_price = models.DecimalField(max_digits=7, decimal_places=2)
+    is_paid = models.BooleanField(default=False)
+    paid_at = models.DateTimeField(auto_now_add=False, null=True, blank=True)
+    is_delivered = models.BooleanField(default=False)
+    delivered_at = models.DateTimeField(auto_now_add=False, null=True, blank=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    def __str__(self):
+        return str(self.created_at)
+
+
+class OrderItem(models.Model):
+    order = models.ForeignKey(Order, on_delete=models.SET_NULL, null=True)
+    product = models.ForeignKey(Product, on_delete=models.SET_NULL, null=True)
+    name = models.CharField(max_length=255)
+    quantity = models.IntegerField(default=0)
+    image = models.CharField(max_length=255)
+
+    def __str__(self):
+        return self.name
+
+
+class ShippingAddress(models.Model):
+    user = models.OneToOneField(User,on_delete=models.CASCADE)
+    order = models.OneToOneField(Order, on_delete=models.CASCADE)
+    address = models.CharField(max_length=255)
+    city = models.CharField(max_length=255)
+    postal_code = models.CharField(max_length=255)
+    country = models.CharField(max_length=255)
+    shipping_price = models.DecimalField(max_digits=7, decimal_places=2)
```

Comment:
I don't think you need this here, as the shipping price is something that should be connected to orders.

Reply:
Done